libs/features/personalization/personalization.js

```diff
@@ -555,6 +556,11 @@ export async function applyPers(manifests) {
   deleteMarkedEls();
 
   const experiments = results.map((r) => r.experiment);
+  if (experiments.length) {
+    const manifestLh = experiments.map((e) => e.manifest.split('/').pop().replace('.json', ''));
+    const variantLh = experiments.map((e) => e.selectedVariantName?.replace('target-', '') || 'default');
+    document.body.dataset.mep = `${variantLh.join('--')}|${manifestLh.join('--')}`;
```

Comment:
Edge case: If there are manifests but no experiments, then `document.body.dataset.mep` will not be defined, which means that there will be a trailing `|` in the analytics string.

Reply:
Well I set it in 3 places.  Line 562, line 543: https://github.com/adobecom/milo/blob/476d83100b3ed1a4e9822bdd891b2aa3c54cb175/libs/features/personalization/personalization.js#L543
and 827 in utils https://github.com/adobecom/milo/blob/476d83100b3ed1a4e9822bdd891b2aa3c54cb175/libs/utils/utils.js#L827

So I think it will set the value correctly if there are no experiments.  But if you think there is a scenario where those don't cover it, I can add an else clause here.

That being said, a trailing `|` is preferred when there is no value.  It still creates the level but with an empty string, thus keeping all of the other levels in their correct place.